media/js/base/global.js

```diff
@@ -76,6 +76,7 @@ function initLangSwitcher() {
             'languageSelected': $language.val(),
             'previousLanguage': previousLanguage
         });
+        $('#lang_form').attr('action', $(location).attr('hash') || '#');
```

Comment:
No other reason than I saw the site uses jquery so I used it :) I will update the patch to use javascript directly.